components/form-builder/app/responses/Dialogs/DialogStates.ts

```diff
@@ -0,0 +1,12 @@
+export enum DialogStates {
```

Comment:
This is breaking the "interface" segregation principle :). I couldn't come up with a good way to extend an enum in TS. Searching around it seems common to use a type with a union. But then an enum would no longer be being used.

Reply:
Curious if this would work

Noting it might not but curious as I said :)

```
export const DialogStates = {
  EDITTING: "EDITTING",
  SENDING: "SENDING",
  SENT: "SENT",
  MIN_ERROR: "MIN_ERROR",
  MAX_ERROR: "MAX_ERROR",
  FORMAT_ERROR: "FORMAT_ERROR",
  DESCRIPTION_EMPTY_ERROR: "DESCRIPTION_EMPTY_ERROR",
  FAILED_ERROR: "FAILED_ERROR",
  UNKNOWN_ERROR: "UNKNOWN_ERROR",
} as const;

export const ResponsesStates = {
  ...DialogStates,
  SOMETHING_ELSE: "SOMETHING_ELSE",
} as const;

export type DialogStatesType = (typeof DialogStates)[keyof typeof DialogStates];

export type ResponsesStatesType = (typeof ResponsesStates)[keyof typeof ResponsesStates];
```

Reply:
When I try this  e.g.
`
const [status, setStatus] = useState<ResponsesStatesType>(ResponsesStatesType.EDITTING);
`
I keep seeing a TS error about
`"'ResponsesStatesType' only refers to a type, but is being used as a value here."`

I could just drop using an enum here.. hmm..

Reply:
Will approve as is and we can play around further with it later.